moabb/evaluations/splitters.py

```diff
@@ -101,3 +110,164 @@ def split(self, y, metadata):
+    cross-validation on data from each subject.
+
+    It assumes that the entire metainformation across all subjects is already loaded.
+
+    Unlike the `CrossSessionEvaluation` class from `moabb.evaluation`, which manages
+    the complete evaluation process end-to-end, this splitter is solely responsible
+    for dividing the data into training and testing sets based on sessions.
+
+    .. image:: images/crosssess.jpg
+        :alt: The schematic diagram of the CrossSession split
+        :align: center
+
+    The inner cross-validation strategy can be changed by passing the
+    `cv_class` and `cv_kwargs` arguments. By default, it uses LeaveOneGroupOut,
+    which performs Leave-One-Session-Out cross-validation.
+
+    Parameters
+    ----------
+    cv_class: cross-validation class, default=LeaveOneGroupOut
+        Inner cross-validation strategy for splitting the sessions of one subject.
+        LeaveOneGroupOut is the most common default.
+    shuffle: bool, default=False
+        Whether to shuffle the session order for each subject. It can only be
+        used when changing the `cv_class` to a class compatible with `shuffle`.
+    random_state: int, RandomState instance or None, default=None
+        Controls the randomness of the inner cross-validation when `shuffle` is True.
+        Pass an int for reproducible output across multiple function calls.
+        For `cv_class` accepting `random_state`, they are provided with a shared rng.
+    cv_kwargs: dict
+        Additional arguments to pass to the inner cross-validation strategy.
+
+    Yields
+    ------
+    train : ndarray
+        The training set indices for that split.
+
+    test : ndarray
+        The testing set indices for that split.
+    """
+
+    def __init__(
+        self,
+        cv_class: type[BaseCrossValidator] = LeaveOneGroupOut,
+        shuffle: bool = False,
+        random_state: int = None,
+        **cv_kwargs,
+    ):
+        self.cv_class = cv_class
+        self.cv_kwargs = cv_kwargs
+        self._cv_kwargs = dict(**cv_kwargs)
+
+        self.shuffle = shuffle
+        self.random_state = random_state
+
+        if not shuffle and random_state is not None:
+            raise ValueError("`random_state` should be None when `shuffle` is False")
+
+        params = inspect.signature(self.cv_class).parameters
+        # We need to check the random_state and shuffle parameters
+        # because for some cv_class, shuffle is not supported
+        # like ShuffleSplit or ShuffleGroupSplit.
+        if shuffle and ("shuffle" not in params and "random_state" not in params):
+            raise ValueError(
+                f"Shuffling is not supported for {cv_class.__name__}. "
+                "Choose a different `cv_class` or use `shuffle=False`."
+                "Example of `cv_class`: `GroupShuffleSplit`: "
+                "CrossSessionSplitter(shuffle=True, random_state=42, cv_class=GroupShuffleSplit)"
+            )
```

Comment:
```suggestion

        params = inspect.signature(self.cv_class).parameters
        if shuffle and "shuffle" not in params:
            raise ValueError(
                f"Shuffling is not supported for {cv_class.__name__}. "
                "Choose a different `cv_class` or use `shuffle=False`."
                "Example of `cv_class`: `GroupShuffleSplit`: "
                "CrossSessionSplitter(shuffle=True, random_state=42, cv_class=GroupShuffleSplit)"
            )

        if not shuffle and "shuffle" in params and random_state is not None:
            raise ValueError(
                "`random_state` should be None when `shuffle` is False for {cv_class.__name__}")
        
```

Reply:
I applied the suggestion, but for the raise of the ValueError, it needs to include this part:  

`and "random_state" not in params.`

Mostly because *Shuffle* classes from sklearn didn't have the parameter shuffle.